src/test/java/com/google/firebase/testing/IntegrationTestUtils.java

```diff
@@ -40,26 +40,25 @@
 import org.apache.http.entity.StringEntity;
 import org.apache.http.impl.client.DefaultHttpClient;
 import org.apache.http.util.EntityUtils;
-import org.json.JSONObject;
 
 public class IntegrationTestUtils {
 
-  private static JSONObject IT_SERVICE_ACCOUNT;
+  private static final String IT_SERVICE_ACCOUNT_PATH = "integration_cert.json";
+  private static final String IT_API_KEY_PATH = "integration_apikey.txt";
+
+  private static GenericJson serviceAccount;
+  private static String apiKey;
   private static FirebaseApp masterApp;
 
-  private static synchronized JSONObject ensureServiceAccount() {
-    if (IT_SERVICE_ACCOUNT == null) {
-      String certificatePath = System.getProperty("firebase.it.certificate");
-      checkArgument(!Strings.isNullOrEmpty(certificatePath),
-          "Service account certificate path not set. Set the "
-              + "firebase.it.certificate system property and try again.");
-      try (InputStreamReader reader = new InputStreamReader(new FileInputStream(certificatePath))) {
-        IT_SERVICE_ACCOUNT = new JSONObject(CharStreams.toString(reader));
+  private static synchronized GenericJson ensureServiceAccount() {
+    if (serviceAccount == null) {
+      try (InputStream stream = new FileInputStream(IT_SERVICE_ACCOUNT_PATH)) {
+        serviceAccount = Utils.getDefaultJsonFactory().fromInputStream(stream, GenericJson.class);
       } catch (IOException e) {
         throw new RuntimeException("Failed to read service account certificate", e);
```

Comment:
Should we make this error more helpful?  Something like "Failed to read service account certificate from &lt;path&gt; see CONTRIBUTING.md for more details."

Reply:
Done